thrift-logger/src/main/java/com/pinterest/singer/client/ThriftLoggerFactory.java

```diff
@@ -56,19 +57,17 @@
   }
 
   /**
-   * Create a default implementation of the ThriftLoggerFactory.
-   * Right now we only support the logback-based thrift logger.
-   * Call this at the start of your server to initialize the thrift logging layer.
+   * Refactoring: Let initialize() method create AuditableLogbackThriftLoggerFactory by default.
+   * AuditableLogbackThriftLoggerFactory will create AuditableLogbackThriftLogger if audit
+   * configurations are used, otherwise it will create LogbackThriftLogger. Call this at the start
+   * of your server to initialize the thrift logging layer.
    */
-  @Deprecated
-  public static synchronized void initialize(
-      File outputDirectory, int rotateThresholdKilobytes) {
+  public static synchronized void initialize(File outputDirectory, int rotateThresholdKilobytes) {
```

Comment:
there is duplicated code among three `initialize` method. can we avoid duplicated code?

Reply:
the third init method: 'initializeAuditLogbackThriftLoggerFactory'  can be removed.   Because these other 2 init methods are using AuditableLogbackThriftLoggerFactory under the hood.   

Will make the change in the update of the PR.

Reply:
can we simplify the two `initialize(...)` methods as follows?  the comments above those two methods are also the same


  /**
   * Refactoring: Let initialize() method create AuditableLogbackThriftLoggerFactory by default.
   * AuditableLogbackThriftLoggerFactory will create AuditableLogbackThriftLogger if audit
   * configurations are used, otherwise it will create LogbackThriftLogger. Call this at the start
   * of your server to initialize the thrift logging layer.
   */
  public static synchronized void initialize(File outputDirectory, int rotateThresholdKilobytes) {
    if (THRIFT_LOGGER_FACTORY_INSTANCE != null) {
      LOGGER.info("Already initialized factory instance. Not initializing another instance.");
      return;
    }

    if (outputDirectory == null) {
      THRIFT_LOGGER_FACTORY_INSTANCE =
          new AuditableLogbackThriftLoggerFactory(outputDirectory, rotateThresholdKilobytes);
    } else {
      THRIFT_LOGGER_FACTORY_INSTANCE =  new AuditableLogbackThriftLoggerFactory();
    }
    LOGGER.info(SUCCESSFUL_INITIALIZATION_MSG + THRIFT_LOGGER_FACTORY_INSTANCE);

    // Add a hook to shutdown loggers on program exit.
    Runtime.getRuntime().addShutdownHook(new Thread() {
      @Override
      public void run() {
        THRIFT_LOGGER_FACTORY_INSTANCE.shutdown();
      }
    });
  }

  public static synchronized void initialize() {
    initialize(null, -1);
  }

Reply:
refactored in the update of the PR.